actions/run.py

```diff
@@ -61,8 +61,29 @@ def encode_dict(in_dict):
         data = urlencode(encode_obj(params))
 
         if http_method == 'POST':
-            response = requests.post(url=url,
-                                     headers=headers, data=data)
+            if files:
+                # we do NOT want Content-Type set in headers here because the content type
+                # set above (application/x-www-form-urlencoded) will NOT work to
+                # upload files. Instead we need requests to create a unique
+                # Content-Type: multipart/formdata; boundary-----XYZ123
+                # This Content-Type and boundary string is unique will be generated
+                # automatically for us if we do no specify a Content-Type.
+                # If we specify headers here with a Content-Type then file uploads
+                # will not work and return mysterious errors
+                headers.pop('Content-Type', None)
+
+                # We're passing data as the params dict instead of
```

Comment:
Slight wording suggestion:

```suggestion
                # We're passing the params dict into the data parameter instead of
```